ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/map.cesium.js

```diff
@@ -366,7 +397,38 @@ module.exports = function CesiumMap(
         },
       })
     },
-    zoomToExtent: function(coords) {},
+    zoomOut: function(opts, done) {},
+    doPanZoom: function(coords, opts = {}) {
+      // debugger
+      const viewRect = createRectangleAroundCoords(coords)
+      map.scene.camera.flyTo({
+        duration: 2.0,
+        destination: getDestinationForVisiblePan(viewRect, map),
+        complete: function() {
+          map.scene.camera.flyTo({
+            duration: 0.5,
+            destination: getDestinationForVisiblePan(viewRect, map)
+          })
+        }
+      })
+    },
+    zoomToExtent: function(coords) {
```

Comment:
This method is unused.

Reply:
What I would do is instead have your component call zoomToExtent directly.  Then you can implement this as
```
 const viewRect = createRectangleAroundCoords(coords)
this.panToRectangle(viewRect)
```

Reply:
That won't work because the openlayers map has an incompatible `zoomToExtent` method, but I'll try that in `doPanZoom` ... (or `zoomToCoords` whatever we want the name to be; but we def need the new method).